src/Microsoft.Health.Dicom.Core/Features/Store/StoreService.cs

```diff
@@ -80,6 +84,7 @@ public class StoreService : IStoreService
     {
         if (instanceEntries != null)
         {
+            _dicomRequestContextAccessor.RequestContext.PartCount = instanceEntries.Count;
```

Comment:
I think this is fine for now, but seeing this mysterious type collect assignments around the code seems weird. Like, is there a place where this sort of computation/interjection makes sense. This is the current pattern though, so I think it's ok for now lol.

Reply:
Yeah, I agree it's a bit odd but I can't think of a singular place that makes sense. I assume you are thinking of all the other fields on the request context used to capture billing as well (BytesTranscoded, IsTranscodeRequested), so it would need to have access to all of the internal data here.

Maybe in the future we could have something after the service layer (StoreService, RetrieveResourceService, RetrieveMetadataService), but before it returns a response object. It could take the stream data & then grab the necessary data to add to the request context. But I haven't fully thought this out so I don't know if it would make it better or worse haha

Reply:
Plus 1, someway to separate billing concern